src/components/Navigation/AuthenticatedNavBar.js

```diff
@@ -0,0 +1,103 @@
+// Local Component Imports
+import Avatar from "@material-ui/core/Avatar";
+import {stringToColor} from "../FeatureViews/Accounts/accountUtils";
+import Tooltip from "@material-ui/core/Tooltip";
+import {fullName} from "../../utils";
+import {closeRegistrationCart} from "../OmouComponents/RegistrationUtils";
+
+const AuthenticatedNavBar = ({toggleDrawer}) => {
+    const dispatch = useDispatch();
+    const history = useHistory();
+    const authUser = useSelector(({auth}) => auth);
+
+    const [mobileOpen, setMobileOpen] = useState(false);
+    const [isMobileSearching, setMobileSearching] = useState(false);
+
+    const handleDrawerToggle = useCallback(() => {
+        toggleDrawer(!mobileOpen);
+        setMobileOpen((open) => !open);
+    }, [mobileOpen, toggleDrawer]);
+
+    const handleLogout = useCallback(() => {
+        closeRegistrationCart();
+        dispatch(logout());
+        history.push("/login");
+    }, [dispatch, history]);
+
+    const handleMobileSearch = useCallback((searchQuery) => {
+        setMobileSearching(searchQuery);
+    }, []);
+
+    const name = fullName(authUser.user);
+
+    return (
+        <AppBar className="OmouBar" position="sticky">
+            <Toolbar>
+                {!isMobileSearching && (
+                    <>
+                        <Hidden lgUp>
+                            <IconButton aria-label="Open Drawer" style={{color: "white"}}
+                                        onClick={handleDrawerToggle}>
+                                <MenuIcon/>
+                            </IconButton>
+                        </Hidden>
+                        <Typography className="title" component={NavLinkNoDup}
+                                    to="/">
+                            omou
+                        </Typography>
+                        <div style={{"flex": 1}}/>
+                    </>
+                )}
+                <Search onMobileType={handleMobileSearch}/>
+                <Tooltip title={`${name}'s Profile`}>
+                    <Avatar
+                        className="avatar"
+                        alt={name}
+                        style={{
+                            "backgroundColor": stringToColor(name),
+                            fontSize: ".9em",
+                            height: "40px", width: "40px",
+                            textDecoration: "none",
+                            border: "2px solid white",
+                            marginRight: "2rem",
+                        }}
+                        component={NavLinkNoDup}
+                        to={`/accounts/${authUser.accountType.toLowerCase()}/${authUser.user.id}`}
+                    >
+                        {name.match(/\b\w/ug).join("")}
+                    </Avatar>
+                </Tooltip>
+                {name === "NeLSoN" &&
+                <Typography className="catsButton" component={NavLinkNoDup} style={{marginLeft: "-2rem"}}
```

Comment:
I would put style on the next line with "to" to make the formatting more balanced

Reply:
lol you've found our easter egg